app/src/main/java/com/example/peterchu/watplanner/coursedetail/CourseDetailPresenter.java

```diff
@@ -58,6 +58,7 @@ public void start() {
             }
 
             Course course = dbHandler.getCourse(courseId);
+            if (course == null) { courseDetailFragment.getActivity().finish(); return; }
```

Comment:
If this happens (course == null), does that mean our app is in a really weird state? If so, maybe we need to log an error or something at least so we know if the app enters this state.